File: player/rb/lib/handler.rb

class PlayerStrategy::Handler
  def name
    ARGV.first
  end

  def competitor_status(competitor)
  end

  def hole_card(card)
  end

  def community_card(card)
  end

  def bet(competitor, bet)
  end

  def bet_request(pot, limits)
    case rand(0..4)
      when 0
        0
      when 1
        limits.to_call + limits.minimum_raise * rand(1..4)
      else
        limits.to_call
    end
  end

  def showdown(comptetior, cards, hand)
  end

  def winner(competitor, amount)
  end
end

File: player/py/lib/handler.py
import sys

class PlayerHandler:

  def name(self):
    return "Peter Python"

  def competitor_status(self, competitor):
    pass

  def hole_card(self, card):
    pass

  def community_card(self, card):
    pass

  def bet(self, competitor, bet):
    pass

  def bet_request(self, pot, limits):
    return 0

  def showdown(self, comptetior, cards, hand):
    pass

  def winner(self, competitor, amount):
    pass

  def shutdown(self):
      sys.exit('Shutting down server')